transform/snowflake-dbt/models/events/nightly/user_events_by_date.sql

```diff
@@ -0,0 +1,240 @@
+
+
+WITH min_active              AS (
+    SELECT
+        user_actual_id as user_id
+      , COALESCE(user_events_telemetry.user_id, IFF(LENGTH(user_events_telemetry.context_server) < 26, NULL, user_events_telemetry.context_server),
+                          IFF(LENGTH(user_events_telemetry.context_traits_userid) < 26, NULL,user_events_telemetry.context_traits_userid),
+                          IFF(LENGTH(user_events_telemetry.context_server) < 26, NULL, user_events_telemetry.context_server)) as server_id
+
+      , MAX(user_actual_role) AS user_actual_role
+      , min(received_at::date) AS min_active_date
+      , CASE WHEN MAX(received_at::date) < CURRENT_DATE
+            THEN CURRENT_DATE ELSE
+            MAX(received_at::date) END AS MAX_ACTIVE_DATE 
+    FROM "ANALYTICS".events.user_events_telemetry
+    WHERE user_actual_id is not null
+    GROUP BY 1, 2),
+
+     dates                   AS (
+         SELECT
+             d.date
+           , m.user_id
+           , m.server_id
+           , m.user_actual_role
+         FROM "ANALYTICS".util.dates      d
+              JOIN min_active m
+                   ON d.date >= m.min_active_date
+                       AND d.date <= m.max_active_date
+         GROUP BY 1, 2, 3, 4
+     ),
+     events                  AS (
+         SELECT
+             d.date
+           , d.user_id
+           , d.server_id                                                                       AS server_id 
+           , MAX(CASE WHEN COALESCE(e.user_actual_role, 'none') in ('system_admin', 'system_admin, system_user', 'system_administrator',
+                                             'admin_user', 'admin', 'system_user, system_admin', 'root_user','super_user',
+                                             'administrator')
```

Comment:
Can we define(`SET`) a variable for the set of roles that we could reuse in more than one statement?

Reply:
I'd have to take a closer look at the code, I only copied it from the last PR and added it back so the dashboard gets updated.